subsys/net/ip/net_mgmt.c

```diff
@@ -237,28 +237,32 @@ static int mgmt_event_wait_call(struct net_if *iface,
 
 	ret = k_sem_take(sync.sync_call, timeout);
 	if (ret == -EAGAIN) {
-		ret = -ETIMEDOUT;
-	} else {
-		if (!ret) {
-			if (raised_event) {
-				*raised_event = sync.raised_event;
-			}
+		net_mgmt_del_event_callback(&sync);
+		return -ETIMEDOUT;
+	}
 
-			if (event_iface) {
-				*event_iface = sync_data.iface;
-			}
+	if (ret < 0) {
+		net_mgmt_del_event_callback(&sync);
+		return ret;
+	}
```

Comment:
could you rewrite the above as such:

```
if (ret < 0) {
     if (ret == -EAGAIN) {
         ret = -ETIMEDOUT;
    }

    net_mgmt_del_event_callback(&sync);
    return ret;
}
```

it's shorter/clearer I think.

Reply:
Rewritten as you have suggested :)

See last force-push